instance/tests/models/test_openedx_database_mixins.py

```diff
@@ -372,6 +375,35 @@ def test_ansible_settings_no_mysql_server(self):
         self.instance.save()
         self.check_mysql_vars_not_set(self.instance)
 
+    @patch('instance.models.mixins.database._get_mysql_cursor', return_value=Mock())
+    @patch('instance.models.mixins.database._drop_database')
+    @patch('instance.models.mixins.database._drop_user')
+    def test_deprovision_mysql(self, mock_drop_user, mock_drop_database, mock_get_cursor):
+        """
+        Test deprovision_mysql does correct calls.
+        """
+        self.instance = OpenEdXInstanceFactory()
+        self.instance.mysql_provisioned = True
+        self.instance.deprovision_mysql()
+        for database in self.instance.mysql_databases:
+            mock_drop_database.assert_any_call(mock_get_cursor(), database["name"])
+            mock_drop_user.assert_any_call(mock_get_cursor(), database["user"])
+        for user in self.instance.global_users:
+            mock_drop_user.assert_any_call(mock_get_cursor(), user)
+
+    @patch('instance.models.mixins.database._get_mysql_cursor', return_value=Mock())
+    @patch('instance.models.mixins.database._drop_database')
+    @patch('instance.models.mixins.database._drop_user')
+    def test_ignore_errors_deprovision_mysql(self, mock_drop_user, mock_drop_database, mock_get_cursor):
+        """
+        Test mysql is set as deprovision when ignoring errors.
+        """
+        self.instance = OpenEdXInstanceFactory()
+        self.instance.mysql_provisioned = True
+        mock_drop_database.side_effect = MySQLError()
```

Comment:
Could we move it to the decorator?
```python
@patch('instance.models.mixins.database._drop_database', side_effect=MySQLError())
```

Reply:
I think it reads better to write it explicitly in the test since that's the point of the test. To handle a `MySQLError`.
But I can move it to the `@patch` decorator.

Reply:
It's nice to clearly separate data preparation from actual testing.

Reply:
I see, let me move it to the `@patch` decorator :)